integration_tests/tests/mod.rs

```diff
@@ -0,0 +1,408 @@
+mod close;
+mod establish;
+mod pay;
+
+use zeekoe::{
+    amount::Amount,
+    customer::{self, database::StateName as CustomerStatus, zkchannels::Command},
+    merchant::{self, zkchannels::Command as _},
+    protocol::ChannelStatus as MerchantStatus,
+};
+use zkabacus_crypto::{CustomerBalance, MerchantBalance};
+
+use crate::common::{customer_cli, merchant_cli, Party, ERROR_FILENAME};
+use anyhow::Context;
+use std::{convert::TryInto, fs::File, io::Read, panic, str::FromStr, time::Duration};
+use structopt::StructOpt;
+use thiserror::Error;
+use zeekoe::customer::database::ClosingBalances;
+
+/// Default balance for establishing channels
+const DEFAULT_BALANCE: u64 = 5;
+/// Default payment; should be less than balance
+const DEFAULT_PAYMENT: u64 = 1;
+
+/// Struct to represent one test: includes a name and a list of operations and outcomes
+#[derive(Debug, Clone)]
+pub struct Test {
+    pub name: String,
+    pub operations: Vec<(Operation, Outcome)>,
+}
+
+/// Set of operations that can be executed by the test harness
+#[allow(unused)]
+#[derive(Debug, Clone, Copy)]
+pub enum Operation {
+    Establish(u64),
+    Pay(u64),
+    PayAll,
+    MutualClose,
+    CustomerClose,
+    MerchantExpiry,
+    Store,
+    Restore,
+    NoOp,
```

Comment:
suggestion: this is not on you, either, but now that we have tests, we don't need this `NoOp` placeholder. I used it when I was testing the overall framework and never removed it. Can you please delete?

Also `Operation`s are definitely used, so the `#[allow(unused)]` here can go.

Reply:
I think this tag is still there because some of the options are still unused. I can make them individual if that's preferred but there are still a quite a few so perhaps the one tag for the whole enum is cleaner?

Reply:
oh, I see. Fine to leave as-is, then!